### PR TITLE
Add shared folder to artifacts

### PR DESCRIPTION
**Pull request by OlegLoewen**
**What this PR does / why we need it**:
Adds shared folder which can be used across the flow steps.

**Release note**:

```improvement operator
Add shared folder for testflow steps.
```

### CODE REVIEW

File: pkg/testmachinery/types.go [hunk 1]
```diff
@@ -26,6 +26,9 @@ const (
 	// TM_KUBECONFIG_PATH is the path where kubeconfigs are mounted to tests.
 	TM_KUBECONFIG_PATH = "/tmp/env/kubeconfig"
 
+	// TM_KUBECONFIG_PATH is the path where kubeconfigs are mounted to tests.
+	TM_SHARED_PATH = "/tmp/shared"
```
Comment (schrodit) [inline pkg/testmachinery/types.go]: I would change this to a testmachinery namespace --> `tmp/tm/shared`

Comment (schrodit) [reply pkg/testmachinery/types.go]: don't know why we had TM_KUBECONFIG_PATH at `tmp/env/kubeconfig`, maybe you could change this path as well to `tmp/tm/kubeconfig`

Comment (OlegLoewen) [reply pkg/testmachinery/types.go]: Done

File: docs/GetStarted.md [hunk 2]
```diff
@@ -117,6 +118,10 @@ It then automatically uploads these documents to an index named like the TestDef
       { "key3": 5 }
     ```
 
+### Shared Folder
+
+Data that is stored in `TM_SHARED_PATH` location, can be accessed from within any testflow step of a the workflow. This is essential if e.g. a test flow step needs to evaluate the output of the previously finished test flow step. This folder also available as an artifact in the Argo UI.
```
Comment (dguendisch) [inline docs/GetStarted.md]:
```suggestion
Data that is stored in `TM_SHARED_PATH` location, can be accessed from within any testflow step of a the workflow. This is essential if e.g. a test flow step needs to evaluate the output of the previously finished test flow step. This folder is also available as an artifact in the Argo UI.
```

File: pkg/testmachinery/types.go [hunk 3]
```diff
@@ -24,7 +24,10 @@ import (
 
 const (
 	// TM_KUBECONFIG_PATH is the path where kubeconfigs are mounted to tests.
-	TM_KUBECONFIG_PATH = "/tmp/env/kubeconfig"
+	TM_KUBECONFIG_PATH = "/tmp/tm/kubeconfig"
+
+	// TM_KUBECONFIG_PATH is the path where kubeconfigs are mounted to tests.
```
Comment (dguendisch) [inline pkg/testmachinery/types.go]: should be an own comment (copy/paste) :)

Comment (OlegLoewen) [reply pkg/testmachinery/types.go]: Done

File: test/controller/testflow/testflow_suite_test.go [hunk 4]
```diff
@@ -177,6 +178,54 @@ var _ = Describe("Testflow execution tests", func() {
 		})
 	})
 
+	Context("sharedFolder", func() {
+		sharedFilePath := fmt.Sprintf("%s/%s", testmachinery.TM_SHARED_PATH, "test")
```
Comment (schrodit) [inline test/controller/testflow/testflow_suite_test.go]: move to test if not needed in other tests

**Pull request merged**